src/darwin/Framework/CHIP/ServerEndpoint/MTRIMDispatch.mm

```diff
@@ -63,49 +83,61 @@ void emberAfClusterInitCallback(EndpointId endpoint, ClusterId clusterId)
 
     void DispatchSingleClusterCommand(const ConcreteCommandPath & aPath, TLV::TLVReader & aReader, CommandHandler * aCommandObj)
     {
-        // TODO: Consider having MTRServerCluster register a
-        // CommandHandlerInterface for command dispatch.  But OTA would need
-        // some special-casing in any case, to call into the existing cluster
-        // implementation.
         using Protocols::InteractionModel::Status;
+
+        if (aPath.mClusterId != OtaSoftwareUpdateProvider::Id) {
```

Comment:
I would not have expected us to get here at all if the cluster is not OtaSoftwareUpdateProvider or the endpoint is not 0, since `InteractionModelEngine::ValidateCommandCanBeDispatched` would return failure for those cases, no?  Why are we adding this check (especially since we are leaving the comment that says they are not needed, though that needs to be updated to use the right function name, probably, since we renamed ServerClusterCommandExists).

Reply:
I mean, this is command dispatch that happens after all the IM checks have been done, no?

Reply:
IM checks are validating by the provider. This code says "I know the provider is set up in a certain way, so I know that I will only get OTA requests". Which is fair, but very coupled.

At the same time, I can just remove this if you think it is wrong/annoying.